### PR TITLE
Implement PayPal credit on checkout

### LINKED ISSUE #292: Offer PayPal Credit on Checkout
Ticket: 595437-zd-woothemes

Currently, plugin only offers PayPal Credit on the cart page, not the checkout. Because of this, customers who use PayPal Credit are not forced to sign up for an account. Even with guest checkout unchecked, customers are sent directly to PayPal Credit and not directed back to sign up for an account.

Please consider adding this feature so our users can take advantage of the PayPal credit option.

This feature is necessary when selling digital and membership based products.

### PR DESCRIPTION
**Pull request by bor0**
Fixes #292.

- Refactor checkout methods (code duplication)
- Implement a separate gateway class for PayPal credit (with a separate logo, title)
- Refactor some methods so that they can be used in the context of woocommerce filter

### CODE REVIEW

**Comment by bor0** [comment]
Please let me know if you have any suggestions to the flow. Right now, PayPal Credit is treated as a separate gateway and is available to be selected in the checkout screen as an option (given PayPal Credit is available in the specific country and is enabled)

**Comment by dwainm** [comment]
@bor0 It is essentially a different service so fine to have it's own "gateway" class. This makes it easier to fix in future. One thing I couldn't test was the checkout, maybe as my account/country doesn't allow for PP credit.

**Comment by bor0** [comment]
@dwainm ah yea, sorry. It's only supported in the US (I had US set as my store location)

File: includes/class-wc-gateway-ppec-checkout-handler.php [hunk 1]
```diff
@@ -598,74 +598,78 @@ protected function is_success( $response ) {
...
+
+	/**
+	 * Handler when buyer is checking out from checkout page.
 	 *
-	 * @throws PayPal_API_Exception
+	 * @param int  $order_id Order ID.
+	 * @param bool $use_ppc  Whether to use PayPal credit.
 	 *
-	 * @param int $order_id Order ID
+	 * @return string Redirect URL.
```
Comment (dwainm) [inline includes/class-wc-gateway-ppec-checkout-handler.php]: Mayb some context on the use_ppc parameter?

**Pull request merged**